### PR TITLE
Submit/x2goclient

### PR DESCRIPTION
**Pull request by dasuxullebt**
A package for the x2go-client. Not sure whether I did everything right, it is my first pull request to nixos.

### CODE REVIEW

**Comment by pSub** [comment]
Could you please rebase your changes against nixpkgs/master?

**Comment by dasuxullebt** [comment]
ok. do I have to make a new pull request now, or just wait?

File: pkgs/applications/networking/remote/x2goclient/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{ fetchurl, stdenv, nxproxy, qt4, openldap, libssh, libXpm, cups, makeWrapper }:
+
+stdenv.mkDerivation rec {
+  version = "4.0.3.1";
+  name = "x2goclient-${version}";
+
+  src = fetchurl {
+    url = "http://code.x2go.org/releases/source/x2goclient/x2goclient-4.0.3.1.tar.gz";
+    md5 = "ef9a20ef96f7c31cc20ece9ebbf1e007";
```
Comment (nckx) [inline pkgs/applications/networking/remote/x2goclient/default.nix]: Usual convention is to include `${name}` in the `url` when possible, so it doesn't also need updating on every version bump. Also, I think sha256 hashes are preferred (easily copied from `nix-prefetch-url`).

File: pkgs/applications/networking/remote/x2goclient/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  preBuild = ''
+    lrelease x2goclient.pro;
+    qmake;
+  '';
+  postInstall = ''
+    mkdir -p $out/bin;
+    cp x2goclient $out/bin;
+    cp -a man $out;
+    wrapProgram "$out/bin/x2goclient" --suffix PATH : "${nxproxy}/bin";
+  '';
```
Comment (nckx) [inline pkgs/applications/networking/remote/x2goclient/default.nix]: Huh. I've never seen semi-colons used like that in shell code before... +1 for personal style? :-)

-1 for readability, I'm afraid.

File: pkgs/development/libraries/libxcomp/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{ fetchgit, stdenv, libX11, autoconf, zlib, libjpeg, libpng12 }:
+
+stdenv.mkDerivation rec {
+  name = "libxcomp";
```
Comment (nckx) [inline pkgs/development/libraries/libxcomp/default.nix]: Why not `name = "libxcomp-${version}"`?

File: pkgs/applications/networking/remote/x2goclient/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  meta = {
+    homepage = "http://x2go.org/";
```
Comment (nckx) [inline pkgs/applications/networking/remote/x2goclient/default.nix]: More of a tip than a hint: plain URIs (without variables in them) can be written without quotes.

File: pkgs/development/libraries/libxcomp/default.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    cd nxcomp/;
+    autoconf;
+  '';
+
+  postInstall = ''
+    mkdir $out/lib
+    cp libXcomp.so libXcomp.so.3 libXcomp.so.3.5.0 $out/lib
+    mkdir $out/include
+    cp NX.h $out/include
+	    '';
```
Comment (nckx) [inline pkgs/development/libraries/libxcomp/default.nix]: Stray tab?

File: pkgs/tools/admin/nxproxy/default.nix [hunk 6]
```diff
@@ -0,0 +1,26 @@
+{ fetchgit, stdenv, libxcomp, autoconf }:
+
+stdenv.mkDerivation rec {
+  name = "nxproxy";
+  version = "3.5.0";
```
Comment (nckx) [inline pkgs/tools/admin/nxproxy/default.nix]: Same question as libxcomp above.

File: pkgs/top-level/all-packages.nix [hunk 7]
```diff
@@ -2051,6 +2053,10 @@ let
 
   nylon = callPackage ../tools/networking/nylon { };
 
+  nxproxy = callPackage ../tools/admin/nxproxy { };
+
+  x2goclient = callPackage ../applications/networking/remote/x2goclient { };
```
Comment (nckx) [inline pkgs/top-level/all-packages.nix]: The `### APPLICATIONS` section looks like a better fit for the client.

**Comment by nckx** [comment]
A `git push -f` is usually fine, with an optional comment here to poke reviewers if needed.

**Comment by nckx** [comment]
Hi @dasuxullebt, are you still working on this?

**Comment by dasuxullebt** [comment]
On 01.03.2015 04:21, Tobias Geerinckx-Rice wrote:
> Hi @dasuxullebt <https://github.com/dasuxullebt>, are you still working
> on this?

Not currently. The computer where I installed NixOS is broken. Maybe I 
will soon try to continue it in a VM, but feel free to continue it.

Regards
CSS

**Comment by nckx** [comment]
Will do.

**Pull request closed without merging**